src/dfmt/formatter.d

```diff
@@ -1834,12 +1836,7 @@ private:
         }
         else
         {
-            // You know what's awesome? Windows can't handle its own line
-            // endings correctly.
-            version (Windows)
-                output.put(current.text.replace("\r", ""));
-            else
-                output.put(current.text);
+            output.put(current.text.lineSplitter.joiner(eolString));
```

Comment:
For a moment I thought this would omit the trailing newline. But `dfmt` always ends the output with a newline even if the input doesn't, so t's good. #551 will detect if this ever changes.

Reply:
instead of doing the short and easy, but memory intensive way, how about adding a helper function that first checks if there are non-conforming line-endings and then runs the lineSplitter + appends it all on it?

Reply:
nvm I just realized these are ranges and should be fine as is